src/state/pools/index.ts

```diff
@@ -33,7 +34,10 @@ export const PoolsSlice = createSlice({
     updatePoolsUserData: (state, action) => {
       const { field, value, sousId } = action.payload
       const index = state.data.findIndex((p) => p.sousId === sousId)
-      state.data[index] = { ...state.data[index], userData: { ...state.data[index].userData, [field]: value } }
+
+      if (index >= 0) {
```

Comment:
This was not causing an issue or anything in production but I noticed with test data (where the sousId was mismatched) this threw an error so I added this to be safe.